src/app/clusters/scenes-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,244 @@
+ */
+
+#include <app/clusters/scenes-server/CodegenIntegration.h>
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/scenes-server/SceneTable.h>
+#include <app/clusters/scenes-server/SceneTableImpl.h>
+#include <app/clusters/scenes-server/ScenesManagementCluster.h>
+#include <app/server/Server.h>
+#include <app/static-cluster-config/ScenesManagement.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/config.h>
+#include <app/util/endpoint-config-api.h>
+#include <data-model-providers/codegen/ClusterIntegration.h>
+#include <data-model-providers/codegen/CodegenDataModelProvider.h>
+
+// Cluster configuration sets values based on this. Ensure config is valid.
+// This means it is NOT sufficient to just set the SCENES_MANAGEMENT_TABLE_SIZE in ZAP, but rather
+// the CHIP_CONFIG must be updated.
+
+#if defined(SCENES_MANAGEMENT_TABLE_SIZE) && SCENES_MANAGEMENT_TABLE_SIZE
+static_assert(chip::scenes::kMaxScenesPerEndpoint >= SCENES_MANAGEMENT_TABLE_SIZE,
+              "ZAP configuration for scenes table should be at most CHIP_CONFIG_MAX_SCENES_TABLE_SIZE");
+#endif
+
+using SceneTable = chip::scenes::SceneTable<chip::scenes::ExtensionFieldSetsImpl>;
+
+namespace chip::app::Clusters::ScenesManagement {
+
+namespace {
+
+class DefaultScenesManagementTableProvider : public ScenesManagementTableProvider
+{
+public:
+    void SetParameters(EndpointId endpointId, uint16_t endpointTableSize)
+    {
+        mEndpointId        = endpointId;
+        mEndpointTableSize = endpointTableSize;
+    }
+    EndpointId GetEndpointId() const { return mEndpointId; }
+
+    ScenesManagementSceneTable * Take() override { return scenes::GetSceneTableImpl(mEndpointId, mEndpointTableSize); }
+    void Release(ScenesManagementSceneTable *) override {}
+
+private:
+    EndpointId mEndpointId      = kInvalidEndpointId;
+    uint16_t mEndpointTableSize = scenes::kMaxScenesPerEndpoint;
+};
+
+constexpr size_t kScenesManagementFixedClusterCount = StaticApplicationConfig::kFixedClusterConfig.size();
+constexpr size_t kScenesManagementMaxClusterCount = kScenesManagementFixedClusterCount + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+LazyRegisteredServerCluster<ScenesManagementCluster> gServers[kScenesManagementMaxClusterCount];
+DefaultScenesManagementTableProvider gTableProviders[kScenesManagementMaxClusterCount];
+
+class IntegrationDelegate : public CodegenClusterIntegration::Delegate
+{
+public:
+    ServerClusterRegistration & CreateRegistration(EndpointId endpointId, unsigned clusterInstanceIndex,
+                                                   uint32_t optionalAttributeBits, uint32_t featureMap) override
+    {
+        uint16_t endpointTableSize = 0;
+        VerifyOrDie(Attributes::SceneTableSize::Get(endpointId, &endpointTableSize) ==
+                    Protocols::InteractionModel::Status::Success);
+
+        const EmberAfCluster * cluster = emberAfFindServerCluster(endpointId, ScenesManagement::Id);
```

Comment:
Should we check for a nullptr here or do we assume we died at the previous check if it were the case?
Maybe we could put a comment for why we don't need a nullptr check here?